### ISSUE TITLE
2019.1 support ?

### ISSUE THREAD

**Issue by mert-dev-acc** [iBicha/ImageEffectGraph#11]
Right now, it appears to be tool is not compatible with Unity 2019.1+ due to changes that they've made.  I'm getting errors like this:

![image](https://user-images.githubusercontent.com/3215742/57565549-b5aa4880-73c8-11e9-885f-b8bb9aecb13b.png)

**Comment by iBicha** [comment]
Thanks for the bug report. Looks like they made a few classes private. 
Will have to implement a workaround, and test if there are any breaking changes in the shader library (I have a feeling that there is)

**Comment by McGunn** [comment]
I'm not upgrading to 2019 simply because this is too important for my project. If you submit a bug report or something complaining about the private classes link it here so we can all go complain with you lol. This should honestly be an official Unity feature.

**Comment by iBicha** [comment]
> If you submit a bug report or something complaining about the private classes link it here so we can all go complain with you lol.

I can't really complain since this was created when these graphics packages were marked in `preview`, which means they are not committed to the current public api and they can make any change to it.

>  This should honestly be an official Unity feature.

I agree. I would love to see this repo go away in favor of having shader graph do this out of the box.

**Comment by iBicha** [comment]
Here's my conclusion:

- Unity made most of classes needed for a custom master node `internal` in version 5 of shadergraph.
- I don't see this change anywhere in their change log 
- I don't see any simple nor clean way to support 2019.x, so for now ImageEffectGraph doesn't support it (added this information on the README)
- If there's a way to support 2019.x that I missed, please do share and/or create a pull request.
- If you need this feature with 2019.x, ask Unity to open up the API, and make needed stuff public (or better yet, ask them for effects with shadergraph, maybe if there's enough demand, they will listen and implement such feature)

As for now, I will keep this issue open, even if there's not much to be done at the moment.
Cheers.

**Comment by mert-dev-acc** [comment]
I'll make this tutorial that I've mentioned a while ago. I'm just not sure what kind of image effect would be nice to show. It is weird that they don't have a support for PostFX. I am open for suggestions on tutorial topic. I'll update here when I do.

**Comment by gavaga** [comment]
If anyone is interested in a temporary workaround for this, it's possible to expose the internals of a unity package by patching the package with an "install" script in your own package. I don't have time to write a PR for this project, but you just need to create a file in the right place with code to the effect of

```C#
using System.Runtime.CompilerServices;
[assembly: InternalsVisibleTo("ImageEffectGraph.Editor")]
```

An example of this done in a (still in-dev) project is [fluviofx](https://github.com/thinksquirrel/fluviofx/blob/master/Install/FluvioFXInstall.cs), which has done the same thing for the VFX Graph package.

**Comment by marioleo** [comment]
With Unity 2019.3b+  now you can use Shadergraph to create post processing effects, using the URP [https://github.com/Unity-Technologies/UniversalRenderingExamples](url) I don't know if the method that they are showing in this example will be the final way to do it, but right now it works.

**Comment by mert-dev-acc** [comment]
@marioleo I am afraid this is not conventional post-processing. Instead, it is more like the replacement shaders via custom renderer features. It is possible to effect objects to be renderer, but nothing else. At least, as far as I know.

**Comment by marioleo** [comment]
Yes I think it's not the definite post processing solution, but I've been testing it, and you can achieve post processing effects manipulating the camera texture with a  shadergraph material,  roughly in a similar way that iBicha was doing it with this repo, currently is the only way (as far as I know) in order to create post processing effects with shadergraph with Unity 2019.3b and the latest version of URP

**Comment by mert-dev-acc** [comment]
Why? I believe Render Features are also exists in LWRP and 2019.2? They didn't even rename the title texts on gifs :D
